### PR TITLE
Remove support for numpy < 1.11.0

### LINKED ISSUE #3427: Dropping NumPy 1.10
Increasingly we discover issues with PRs after merging them on NumPy 1.10, which we then either need to fix ourselves or hope the submitter of the PR will kindly fix the issue. While these issues can normally be solved by skipping, adding compat functions, etc., it starts to raise the question: how long do we want to support NumPy 1.10. As of January NumPy 1.14 was relased and has since had 2 patch releases. The list of NumPy 1.15 issue is not too long either (roughly two thirds have already been resolved). To me this is a pretty good case for dropping, but am interested to hear other thoughts.

If we do feel strongly about keeping NumPy 1.10, could we discuss adding it as part of the PR builds to cutdown on strain caused by the issues mentioned above?

### PR DESCRIPTION
**Pull request by jcrist**
Downstream libraries have bumped the minimum numpy version they support, and we should too.

Fixes #3427.

### CODE REVIEW

File: dask/array/ma.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
                    asanyarray, atop)
 
 
-if LooseVersion(np.__version__) < '1.11.0':
-    raise ImportError("dask.array.ma requires numpy >= 1.11.0")
+if LooseVersion(np.__version__) < '1.11.2':
+    raise ImportError("dask.array.ma requires numpy >= 1.11.2")
```
Comment (jcrist) [inline dask/array/ma.py]: There was something that got fixed between 1.11.1 and 1.11.2 that allows the sparse/masked chunks to work. I didn't bother diagnosing it as 1.11.2 is also several versions back, just changed the version requirement.

**Comment by mrocklin** [comment]
cc @shoyer as he seems plugged in here

**Comment by jcrist** [comment]
Seeing a failure due to a test added in #3370. Added a case around it checking if the numpy `__array_ufunc__` protocol is supported, but there *may* be other spots where this is needed (none that show up in tests).

**Comment by jcrist** [comment]
Merging later today if no comment.

**Comment by jakirkham** [comment]
Thanks @jcrist. :)

**Pull request merged**